src/Features/CSharp/Portable/GenerateMember/GenerateVariable/CSharpGenerateVariableService.cs

```diff
@@ -33,7 +32,18 @@ protected override bool IsExplicitInterfaceGeneration(SyntaxNode node)
             => node is PropertyDeclarationSyntax;
 
         protected override bool IsIdentifierNameGeneration(SyntaxNode node)
-            => node is IdentifierNameSyntax identifier && !identifier.Identifier.CouldBeKeyword();
+            => node is IdentifierNameSyntax identifierName && !IsProbablySyntacticConstruct(identifierName.Identifier);
+
+        private static bool IsProbablySyntacticConstruct(SyntaxToken token)
+        {
+            var contextualKind = SyntaxFacts.GetContextualKeywordKind(token.ValueText);
```

Comment:
Is `ValueText` intended? If `@` prefix is used (e.g., `@var`), then `ValueText` will be `var`, while `Text` will be `@var`. I assume we should return false here for `@var`. It would be nice to have a test confirming the behavior, and make sure it fails when `Text` is changed to `ValueText` or vice versa

Reply:
Previously insted of this method `CouldBeKeyword` was used, and `ValueText` was used in its implementation. I checked, and it turned out, that the codefix itself ignore `@` prefix. So if I use `Text` here the codefix will be present, but it will create, for instance, `var` variable instead of `@var`. There are 2 possible ways to solve it: leave current implementation, so codefix won't be suggested to prefixed identifiers like `@var`, or change the behaviour of the codefix itself and do not ignore `@` prefix when creating variable. IMO first variant is preferable. @CyrusNajmabadi Am I right?

Reply:
great comment.